File: DSA/A1/Day-7/P1.cpp
/*
You are given an integer array pref of size n.
The array pref is created from another array arr using this rule:
pref[i] = arr[0] ^ arr[1] ^ ... ^ arr[i]   (for i = 0 to n-1)

Here, ^ means bitwise XOR.

Your task is to find and return the original array arr.

Constraints:
------------------
1 <= pref.length <= 105
0 <= pref[i] <= 106

Input Format:
---------------------
First line → integer n → the size of pref
Second line → n space-separated integers → elements of pref

Output Format:
-----------------------
Print the original array arr in array format (with brackets and commas)

Sample Input:
---------------
5
5 2 0 3 1

Sample Output:
--------------
[5, 7, 2, 3, 2]

Explanation:
-----------------
arr[0] = pref[0] = 5
arr[1] = pref[0] ^ pref[1] = 5 ^ 2 = 7
arr[2] = pref[1] ^ pref[2] = 2 ^ 0 = 2
arr[3] = pref[2] ^ pref[3] = 0 ^ 3 = 3
arr[4] = pref[3] ^ pref[4] = 3 ^ 1 = 2

*/

File: DSA/Elite/Day-12/P1.java
/*
A grid of light bulbs is given, represented as a matrix of size rows x cols, 
where each cell contains either 0 (off) or 1 (on).

Your task is to turn all the light bulbs off (0) by following the toggle rule:
    - In each step, you can choose either an entire row or an entire column 
    and toggle all its elements (change 0 to 1 and 1 to 0).
    
At the end, if all light bulbs are turned off, print true, otherwise print false.


Input Format
-------------
Line-1: Read two integers rows and cols(space separated).
Line-2: Read the matrix of dimension rows X cols.

Output Format
--------------
Print a boolean result.



Sample input-1:
---------------
5 5
0 0 1 0 0
0 0 1 0 0
1 1 0 1 1
0 0 1 0 0
0 0 1 0 0

Sample output-1:
----------------
true

Explanation:
------------
0 0 1 0 0          0 0 1 0 0           0 0 0 0 0
0 0 1 0 0   row-3  0 0 1 0 0   cols-3  0 0 0 0 0
1 1 0 1 1   --->   0 0 1 0 0   --->    0 0 0 0 0
0 0 1 0 0          0 0 1 0 0           0 0 0 0 0
0 0 1 0 0          0 0 1 0 0           0 0 0 0 0 


Sample input-2
--------------
2 2
1 1
0 1

Sample output-2
---------------
false
*/

import java.util.*;


class Main{
    public static void main(String args[]){
        Scanner sc=new Scanner(System.in);
        int r=sc.nextInt();
        int c=sc.nextInt();
        int arr[][]=new int[r][c];
        for(int i=0;i<r;i++){
            for(int j=0;j<c;j++){
                arr[i][j]=sc.nextInt();
            }
        }
        boolean flag=true;
        for(int i=1;i<r;i++){
            int curr=arr[0][0]^arr[i][0];
            for(int j=1;j<c;j++){
                if((arr[i][j]^arr[0][j])!=curr){
                    flag=false;
                    break;
                }
            }
        }
        System.out.println(flag);
    }
}